sql/src/test/java/ddlog/JooqProviderTest.java

```diff
@@ -0,0 +1,86 @@
+/*
+ * Copyright 2018-2020 VMware, Inc. All Rights Reserved.
+ * SPDX-License-Identifier: BSD-2
+ */
+package ddlog;
+
+import com.vmware.ddlog.DDlogJooqProvider;
+import com.vmware.ddlog.ir.DDlogProgram;
+import com.vmware.ddlog.translator.Translator;
+import ddlogapi.DDlogAPI;
+import ddlogapi.DDlogCommand;
+import ddlogapi.DDlogException;
+import ddlogapi.DDlogRecCommand;
+import ddlogapi.DDlogRecord;
+import org.jooq.DSLContext;
+import org.jooq.Record;
+import org.jooq.Result;
+import org.jooq.impl.DSL;
+import org.jooq.tools.jdbc.MockConnection;
+import org.jooq.tools.jdbc.MockDataProvider;
+import org.junit.Test;
+
+import java.io.BufferedWriter;
+import java.io.File;
+import java.io.FileWriter;
+import java.io.IOException;
+import java.util.ArrayList;
+import java.util.List;
+
+import static org.junit.Assert.assertArrayEquals;
+import static org.junit.Assert.assertEquals;
+
+public class JooqProviderTest {
+
+    @Test
+    public void testddlog() throws IOException, DDlogException {
+        final Translator t = new Translator(null);
+        final String s1 = "create table hosts (id integer, capacity integer)";
+        final String v2 = "create view hostsv as select distinct * from hosts";
+        final String v1 = "create view good_hosts as select distinct * from hosts where capacity < 50";
+        t.translateSqlStatement(s1);
+        t.translateSqlStatement(v2);
+        t.translateSqlStatement(v1);
+        final DDlogProgram dDlogProgram = t.getDDlogProgram();
+        writeProgramToFile(dDlogProgram.toString());
+        DDlogAPI.compileDDlogProgram("/tmp/program.dl", true, "../lib", "./lib");
+        DDlogAPI.loadDDlog();
+
+        final DDlogAPI dDlogAPI = new DDlogAPI(1, null, true);
+        final int numInserts = 5;
+        for (int i = 0; i < numInserts; i++) {
+            final DDlogRecord rec = new DDlogRecord(i);
+            final DDlogRecord cap = new DDlogRecord(20);
+            final DDlogRecord struct = DDlogRecord.makeStruct("Thosts", rec, cap);
+            final int id = dDlogAPI.getTableId("Rhosts");
+            final DDlogRecCommand command = new DDlogRecCommand(DDlogCommand.Kind.Insert, id, struct);
```

Comment:
Don't you want to insert everything in one transaction? Should be significantly more efficient.

Reply:
This was just for tests. I'll fix it.